### PR TITLE
Refactor manifest tests (#1430)

### PR DESCRIPTION
**Pull request by jessebrennan**
This wasn't prescribed work, but it will make working on #1430 easier.

I think chaining #1430 on this branch will be easier since hopefully this stuff is mostly uncontroversial, despite more invasive changes, and I expect #1430 to take a while.

### CODE REVIEW

File: test/service/test_manifest.py [hunk 1]
```diff
@@ -648,481 +637,460 @@ def test_bdbag_manifest_for_redundant_entries(self):
...
-
-            filters = {'project': {'is': ['Mouse Melanoma']}}
-            response = self._get_manifest(ManifestFormat.full, filters)
-            self.assertEqual(200, response.status_code, 'Unable to download manifest')
-            # Cannot use response.iter_lines() because of https://github.com/psf/requests/issues/3980
-            lines = response.content.decode('utf-8').splitlines()
-            tsv_file2 = csv.reader(lines, delimiter='\t')
-            fieldnames2 = set(next(tsv_file2))
+            fieldnames.append(set(next(tsv_file1)))
+        intersection = reduce(operator.and_, fieldnames)
```
Comment (hannes-ucsc) [inline test/service/test_manifest.py:1089]:
```suggestion
        intersection = reduce(set.intersection, fieldnames)
```

File: test/service/test_manifest.py [hunk 2]
```diff
@@ -648,481 +637,460 @@ def test_bdbag_manifest_for_redundant_entries(self):
...
+        symmetric_diff = reduce(operator.xor, fieldnames)
```
Comment (hannes-ucsc) [inline test/service/test_manifest.py:1090]:
```suggestion
        symmetric_diff = reduce(set.intersection, fieldnames)
```

File: src/azul/service/elasticsearch_service.py [hunk 3]
```diff
@@ -65,6 +65,9 @@ def __init__(self, missing_index: str):
         super().__init__(f'Index `{missing_index}` was not found')
 
 
+SourceFilters = List[str]
```
Comment (hannes-ucsc) [inline src/azul/service/elasticsearch_service.py:68]: Corresponding commit title could be improved. You're not just moving the alias. And it's *alias*, not *type definition*. https://docs.python.org/3/library/typing.html#type-aliases

File: test/service/test_manifest.py [hunk 4]
```diff
@@ -96,6 +98,22 @@ def _get_manifest_url(self, format_: ManifestFormat, filters: JSON) -> Tuple[str
         return service.get_manifest(format_, filters)
 
 
+def manifest_decorator(test):
+    @mock_sts
+    @mock_s3
+    def wrapper(self, *args, **kwargs):
```
Comment (hannes-ucsc) [inline test/service/test_manifest.py:104]:
```suggestion
def manifest_test(test_method):
    """
    A decorator for test methods that test manifest functionality.
    """
    @mock_sts
    @mock_s3
    def wrapper(self, *args, **kwargs):
```

File: test/service/test_manifest.py (hunk 2 above)
Comment (jessebrennan) [reply test/service/test_manifest.py:1085]: Assuming you meant `set.symmetric_difference` here.

File: src/azul/service/elasticsearch_service.py (hunk 3 above)
Comment (jessebrennan) [reply src/azul/service/elasticsearch_service.py:68]: I renamed that commit, but the rest of the changes are fixups.

**Pull request merged**